itest/lnd_channel_backup_test.go

```diff
@@ -1672,3 +1954,383 @@ func assertDLPExecuted(ht *lntest.HarnessTest,
+			ht.AssertNumPendingSweeps(dave, 2)
+		} else {
+			ht.AssertNumPendingSweeps(dave, 1)
+		}
+
+		// Assert the sweeping tx is mined.
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Funds are now in Dave's wallet.
+		ht.AssertNumPendingForceClose(dave, 0)
+	}
+
+	// We query Dave's balance to make sure it increased after the channel
+	// closed. This checks that he was able to sweep the funds he had in
+	// the channel.
+	daveBalResp := dave.RPC.WalletBalance()
+	daveBalance := daveBalResp.ConfirmedBalance
+	require.Greater(ht, daveBalance, daveStartingBalance,
+		"balance not increased")
+
+	ht.AssertNodeNumChannels(dave, 0)
+}
+
+// assertFundsRecoveredStartCarol asserts that funds are recovered after Dave
+// running scbforceclose procedure. Force close transaction is created from
+// Dave's SCB backup and broadcasted by the caller of this function. Carol is
+// started after the force close transaction is mined.
+func assertFundsRecoveredStartCarol(ht *lntest.HarnessTest,
+	carol *node.HarnessNode, carolStartingBalance int64,
+	dave *node.HarnessNode, daveStartingBalance int64,
+	commitType lnrpc.CommitmentType, startCarol func() error) {
+
+	ht.Helper()
+
+	// Increase the fee estimate so that the following force close tx will
+	// be cpfp'ed.
+	ht.SetFeeEstimate(30000)
+
+	// Upon reconnection, the nodes should detect that Dave is out of sync.
+	// Carol should force close the channel using her latest commitment.
+	// For scbforceclose case: we detect the manually broadcasts Dave's tx.
+	ht.AssertNumTxsInMempool(1)
+
+	// Dave should also consider the channel "waiting close", as he noticed
+	// the channel was out of sync, and is now waiting for a force close to
+	// hit the chain.
+	ht.AssertNumWaitingClose(dave, 1)
+
+	// Restart Dave to make sure he is able to sweep the funds after
+	// shutdown.
+	ht.RestartNode(dave)
+
+	// Generate a single block, which should confirm the closing tx.
+	ht.MineBlocksAndAssertNumTxes(1, 1)
+	blocksMined := uint32(1)
+
+	// Start Carol after mining closing tx if requested. It is done here to
+	// ensure that she does not publish her version of force close tx.
+	require.NoError(ht, startCarol())
+	ht.EnsureConnected(carol, dave)
+
+	// Dave should consider the channel pending force close (since he is
+	// waiting for his sweep to confirm).
+	ht.AssertNumPendingForceClose(dave, 1)
+
+	// Carol is considering it "pending force close", as we must
+	// wait before she can sweep her outputs.
+	ht.AssertNumPendingForceClose(carol, 1)
+
+	if commitType == lnrpc.CommitmentType_SCRIPT_ENFORCED_LEASE {
+		// Dave should sweep his anchor only, since he still has the
+		// lease CLTV constraint on his commitment output. We'd also
+		// see Carol's anchor sweep here.
+
+		// Both Dave and Carol should have an anchor sweep request.
+		// Note that they cannot sweep them as these anchor sweepings
+		// are uneconomical.
+		ht.AssertNumPendingSweeps(dave, 1)
+
+		// Carol has two sweeps.
```

Comment:
can you add which sweep Carol has here in addition to the anchor ?

Reply:
Done. 

```
// Carol has two sweeps: an anchor and her balance from the
// channel. Carol's funds are available immediately, as they are
// not timelocked.
```